File: app/src/main/java/com/babestudios/companyinfouk/ui/search/SearchFilterAdapter.java
package com.babestudios.companyinfouk.ui.search;


import android.content.Context;
import android.support.annotation.NonNull;
import android.support.graphics.drawable.VectorDrawableCompat;
import android.support.v4.content.ContextCompat;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.babestudios.companyinfouk.R;
import com.babestudios.companyinfouk.utils.ResourceHelper;


public class SearchFilterAdapter extends ArrayAdapter<String> {
	private String[] mTexts;
	private boolean isDarkTheme;
	private Context context;
	private int verticalDropdownPadding;

	public SearchFilterAdapter(Context context, String[] texts, boolean isDarkTheme) {
		super(context, R.layout.search_filter_spinner_item, texts);
		this.isDarkTheme = isDarkTheme;
		mTexts = texts;
		this.context = context;
		verticalDropdownPadding = getContext().getResources().getDimensionPixelSize(R.dimen.view_margin_small);
	}

	@Override
	public View getDropDownView(int position, View convertView, @NonNull ViewGroup parent) {
		DropDownViewHolder holder;
		if (convertView == null) {
			LayoutInflater inflater = LayoutInflater.from(parent.getContext());
			convertView = inflater.inflate(R.layout.search_filter_spinner_dropdown_item, parent, false);
			holder = new DropDownViewHolder();
			holder.dropdown_root = convertView.findViewById(R.id.dropdown_item_root);
			holder.dropdown_text_view = (TextView) convertView.findViewById(R.id.dropdown_item_text_view);
			convertView.setTag(holder);
		} else {
			holder = (DropDownViewHolder) convertView.getTag();
		}
		int bottomPadding = 0;
		int topPadding = 0;
		if (position == 0) {
			topPadding = verticalDropdownPadding;
		}
		if (position == (getCount() - 1)) {
			bottomPadding = verticalDropdownPadding;
		}

		if (isDarkTheme) {
			holder.dropdown_root.setBackgroundColor(ContextCompat.getColor(context, android.R.color.white));
		}

		holder.dropdown_root.setPadding(0, topPadding, 0, bottomPadding);
		holder.dropdown_text_view.setText(mTexts[position]);

		return convertView;
	}

	@NonNull
	@Override
	public View getView(int position, View convertView, @NonNull ViewGroup parent) {
		SpinnerItemViewHolder holder;
		if (convertView == null) {
			LayoutInflater inflater = LayoutInflater.from(parent.getContext());
			convertView = inflater.inflate(R.layout.search_filter_spinner_item, parent, false);
			holder = new SpinnerItemViewHolder();
			holder.spinner_item_root = convertView.findViewById(R.id.spinner_item_root);
			holder.spinner_item_icon = (ImageView) convertView.findViewById(R.id.spinner_item_icon);
			holder.spinner_item_text_view = (TextView) convertView.findViewById(R.id.spinner_item_text_view);
			if (isDarkTheme) {
				holder.spinner_item_text_view.setTextColor(ContextCompat.getColor(context, android.R.color.white));
				holder.spinner_item_icon.setImageDrawable(ResourceHelper.tintVectorDrawableCompat(context, VectorDrawableCompat.create
						(context.getResources(), R.drawable.arrow_drop_down_vector, context.getTheme()), android.R.color.white));
			}
			convertView.setTag(holder);
		} else {
			holder = (SpinnerItemViewHolder) convertView.getTag();
		}


		holder.spinner_item_text_view.setText(mTexts[position]);
		return convertView;
	}

	private static class DropDownViewHolder {
		View dropdown_root;
		TextView dropdown_text_view;
	}

	private static class SpinnerItemViewHolder {
		View spinner_item_root;
		ImageView spinner_item_icon;
		TextView spinner_item_text_view;
	}
}

File: app/src/main/java/com/babestudios/companyinfouk/ui/search/SearchFilterAdapter.kt
package com.babestudios.companyinfouk.ui.search


import android.content.Context
import android.support.graphics.drawable.VectorDrawableCompat
import android.support.v4.content.ContextCompat
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView

import com.babestudios.companyinfouk.R
import com.babestudios.companyinfouk.utils.ResourceHelper


class SearchFilterAdapter(context: Context, private val mTexts: Array<String>, private val isDarkTheme: Boolean) : ArrayAdapter<String>(context, R.layout.menu_spinner_item, mTexts) {
	private val verticalDropdownPadding: Int = context.resources.getDimensionPixelSize(R.dimen.view_margin_small)

	override fun getDropDownView(position: Int, convertView: View?, parent: ViewGroup): View {
		var convertViewCopy = convertView
		val holder: DropDownViewHolder
		if (convertViewCopy == null) {
			val inflater = LayoutInflater.from(parent.context)
			convertViewCopy = inflater.inflate(R.layout.menu_spinner_dropdown_item, parent, false)
			holder = DropDownViewHolder()
			holder.llDropdownItemRoot = convertViewCopy!!.findViewById(R.id.llDropdownItemRoot)
			holder.lblDropdownItem = convertViewCopy.findViewById<View>(R.id.lblDropdownItem) as TextView
			convertViewCopy.tag = holder
		} else {
			holder = convertViewCopy.tag as DropDownViewHolder
		}
		var bottomPadding = 0
		var topPadding = 0
		if (position == 0) {
			topPadding = verticalDropdownPadding
		}
		if (position == count - 1) {
			bottomPadding = verticalDropdownPadding
		}

		if (isDarkTheme) {
			holder.llDropdownItemRoot?.setBackgroundColor(ContextCompat.getColor(context, android.R.color.white))
		}

		holder.llDropdownItemRoot?.setPadding(0, topPadding, 0, bottomPadding)
		holder.lblDropdownItem?.text = mTexts[position]

		return convertViewCopy
	}

	override fun getView(position: Int, convertView: View?, parent: ViewGroup): View {
		var convertViewCopy = convertView
		val holder: SpinnerItemViewHolder
		if (convertViewCopy == null) {
			val inflater = LayoutInflater.from(parent.context)
			convertViewCopy = inflater.inflate(R.layout.menu_spinner_item, parent, false)
			holder = SpinnerItemViewHolder()
			holder.llSpinnerItemRoot = convertViewCopy!!.findViewById(R.id.llSpinnerItemRoot)
			holder.ivSpinnerItem = convertViewCopy.findViewById<View>(R.id.ivSpinnerItem) as ImageView
			holder.lblSpinnerItem = convertViewCopy.findViewById<View>(R.id.lblSpinnerItem) as TextView
			holder.lblSpinnerItem?.setTextColor(ContextCompat.getColor(context, android.R.color.white))
			val vector = VectorDrawableCompat.create(context.resources, R.drawable.ic_arrow_drop_down, context.theme)
			vector?.let {
				holder.ivSpinnerItem?.setImageDrawable(ResourceHelper.tintVectorDrawableCompat(context, it, android.R.color.white))
			}
			convertViewCopy.tag = holder
		} else {
			holder = convertViewCopy.tag as SpinnerItemViewHolder
		}


		holder.lblSpinnerItem?.text = mTexts[position]
		return convertViewCopy
	}

	private class DropDownViewHolder {
		internal var llDropdownItemRoot: View? = null
		internal var lblDropdownItem: TextView? = null
	}

	private class SpinnerItemViewHolder {
		internal var llSpinnerItemRoot: View? = null
		internal var ivSpinnerItem: ImageView? = null
		internal var lblSpinnerItem: TextView? = null
	}
}
